nconnect.go

```diff
@@ -30,36 +28,45 @@ import (
 	"github.com/nknorg/nkngomobile"
 	"github.com/nknorg/tuna/filter"
 	"github.com/nknorg/tuna/geo"
+	"github.com/nknorg/tuna/types"
 	"gopkg.in/natefinch/lumberjack.v2"
 )
 
 const (
 	mtu = 1500
 )
 
-func Run(opts *config.Opts) {
-	err := (&opts.Config).SetPlatformSpecificDefaultValues()
-	if err != nil {
-		log.Fatal(err)
-	}
+type nconnect struct {
+	opts    *config.Opts
+	account *nkn.Account
 
-	if opts.Version {
-		fmt.Println(config.Version)
-		os.Exit(0)
-	}
+	walletConfig *nkn.WalletConfig
+	clientConfig *nkn.ClientConfig
+	tunnelConfig *tunnel.Config
+	ssConfig     *ss.Config
+	persistConf  *config.Config
+
+	adminClientCache *admin.Client
+	remoteInfoCache  *admin.GetInfoJSON
+
+	tunnel   *tunnel.Tunnel
+	tunaNode *types.Node // It is used to connect specific tuna node, mainly is for testing.
+}
 
-	if opts.Client == opts.Server {
```

Comment:
IMO it's better to do this verification inside `NewNconnect` because other people might call this function too

Reply:
Yes, restored.